src/ui/UserInterface.java

```diff
@@ -435,19 +437,44 @@ public void checkOut() {
 		}
 	}
 
+	/**
+	 * Processes a new shipment to the grocery store fulfilled by a vendor.
+	 */
 	public void processShipment() {
+		do {
+			String orderNumber = getString("Enter outstanding order number: ");
+			if (!groceryStore.orderIdExists(orderNumber)) {
+				System.out.println("The order number is not on file.");
+				return;
+			}
+			if (!groceryStore.orderIsOutstanding(orderNumber)) {
+				System.out.println("The order " + orderNumber + " has already been processed.");
+				return;
+			}
+			// following portion is carried out only if the order number exists and order is
+			// outstanding
+			// order number is loaded into the request instance
+			Request.instance().setOrderId(orderNumber);
+			// order is processed by GroceryStore's processOrder() method
+			Result result = groceryStore.processOrder(Request.instance());
+			if (result.getResultCode() == Result.ACTION_SUCCESSFUL) {
+				System.out.println("Order " + orderNumber + "successfully processed:");
+				System.out.println("Product " + result.getProductId() + ", '" + result.getProductName()
+						+ "', has a new stock quantity of " + result.getProductStockOnHand() + " unit(s).");
+			} else {
+				System.out.println("The processing of order " + orderNumber + " has failed.");
```

Comment:
Should there be a Result.ACTION_FAILED here?

Reply:
ACTION_FAILED only happens if the order is non-existent or fulfilled already. The GroceryStore's method takes care of it already, in case the UI method failed to include that. Which it does. The action will only fail if the order is "non-existent or fulfilled already", like I said before. That's already filtered earlier.

Reply:
Supposedly, the action would only fail, if the order number didn't exist OR if the order wasn't outstanding, but I could add the else to ACTION_FAILED and perhaps else would be "other" error.

Reply:
Well as long as the result code is being set somewhere when the shipment isn't processed that's what's important. Otherwise looks good. 

Only other note: I really liked that you included the link to the PR in the discord message. Was very easy to get to the review!

Reply:
The safeguard is in processShipment() in UI. The only reason I include another safeguard in GroceryStore's processOrder() is if the method was used independently. To make sure the app does not throw an exception. Which, btw., everybody (incl. me) should implement in all of their methods, to make them fool proof. Not just think, "oh, this has already been taken care of...". If it's a new method, a new safeguard should be applied...!